### PR TITLE
improving process_results

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: platypus/parse.py [hunk 1]
```diff
@@ -263,20 +270,19 @@ def process_results(percentage_ids, alignment_lengths, percentage_ids_other,
...
+                results[i]['summary_fh'].write('%s\t%s\t%s\n' % (
+                    seq_name, subject_id_a, subject_id_b))
                 db_seqs_counts_a[subject_id_a] += 1
                 db_seqs_counts_b[subject_id_b] += 1
             elif vals['a']['bit_score'] > vals['b']['bit_score']:
                 if not subject_id_b:
                     results[i]['perfect_interest'] += 1
-                    results[i]['summary'].append('%s\t%s\t' % (seq_name,
-                                                               subject_id_a))
+                    results[i]['summary_fh'].write('%s\t%s\t\n' % (
```
Comment (wasade) [inline platypus/parse.py]: the header on line 239 has 3 columns but this is only writing two. Would it be possible to write an explicit null value so the resulting file is not jagged?

File: platypus/parse.py [hunk 2]
```diff
@@ -263,20 +270,19 @@ def process_results(percentage_ids, alignment_lengths, percentage_ids_other,
...
+                        seq_name, subject_id_a))
                 db_seqs_counts_a[subject_id_a] += 1
             else:
                 results[i]['db_other'] += 1
-                results[i]['summary'].append('%s\n\t%s' % (seq_name, ''))
+                results[i]['summary_fh'].write('%s\n\t%s\n' % (seq_name, ''))
```
Comment (wasade) [inline platypus/parse.py]: same here, but this is confusing, it looks like it is writing out:

```text
foo
\t<NOTHING>
```

Comment (wasade) [inline platypus/parse.py]: oh, is this a record break? if so, why str format with the `''`?

File: platypus/parse.py [hunk 3]
```diff
@@ -217,6 +217,8 @@ def process_results(percentage_ids, alignment_lengths, percentage_ids_other,
         database.
     best_hits : dict
         A dictionary with the best hits found in the databases.
+    output_dir : str
+        Name of the output file path.
```
Comment (ElDeveloper) [inline platypus/parse.py]: file path -> directory

Comment (ElDeveloper) [inline platypus/parse.py]: Or better yet, "file path to the output directory"?

File: platypus/parse.py (hunk 1 above)
Comment (ElDeveloper) [inline platypus/parse.py]: Results are not gonna be jagged are they?

File: tests/test_parse.py [hunk 4]
```diff
@@ -201,19 +202,17 @@ def test_process_results(self):
                       'alg_length': 900}}]
         }
 
-        out_results = process_results([0.80], [50], [0.30], [30], best_hits)
+        out_results = process_results([0.80], [50], [0.30], [30], best_hits,
+                                      self.base)
+        # removing the sumout_resultsmary pointer so we don't need to test
```
Comment (ElDeveloper) [inline tests/test_parse.py]: I don't remember this anymore, but what's `sumout_resultsmary`?

File: platypus/parse.py (hunk 1 above)
Comment (antgonza) [inline platypus/parse.py]: You are right, it expects 3 values: seq_id best_A best_B. In this case, there is no best_B. Thus, just adding a new line is fine cause there is no value there. However, I can add something more specific but not sure what.

The results shouldn't be jagged. BTW there is a test that checks that the resulting files are the same.

File: platypus/parse.py (hunk 2 above)
Comment (antgonza) [inline platypus/parse.py]: Here there is no result for either the first or second. I guess it was a copy/paste, will change.

File: tests/test_parse.py (hunk 4 above)
Comment (antgonza) [inline tests/test_parse.py]: It used to be an array where each element had the values that now are being written directly to a file. Now, it's a pinter to an open file, thus not testing.

**Comment by antgonza** [comment]
@ElDeveloper @wasade ready for another pass ... thanks!

File: platypus/parse.py [hunk 5]
```diff
@@ -224,19 +226,24 @@ def process_results(percentage_ids, alignment_lengths, percentage_ids_other,
...
+    summary_fh = {}
 
     iter_a = product(percentage_ids, alignment_lengths)
     iter_b = product(percentage_ids_other, alignment_lengths_other)
+
     for (perc_id_a, aln_len_a), (perc_id_b, aln_len_b) in izip(iter_a, iter_b):
         filename = "p1_%d-a1_%d_p2_%d-a2_%d" % (perc_id_a, aln_len_a,
                                                 perc_id_b, aln_len_b)
+        summary_filename = join(output_dir, "summary_" + filename + ".txt")
+        summary_fh = open(summary_filename, 'w')
```
Comment (ElDeveloper) [inline platypus/parse.py]: Hmmm, I just noticed that this file handle is not being closed, or am I missing something.

Comment (antgonza) [inline platypus/parse.py]: That's right, it's not and I think is fine cause their will be closed once the program finishes. If we want to close them, we will need to put a for loop at the end of this one to close them. Should I do that?

**Comment by ElDeveloper** [comment]
Just :point_up: comment.

File: platypus/parse.py (hunk 5 above)
Comment (ElDeveloper) [inline platypus/parse.py]: I think this would be a good idea specially, if the number of files grows, we might run out of file handles :scream:

**Pull request merged**